### ISSUE TITLE
Enhance the accessibility of `Accordion` (`@yamada-ui/accordion`) component

### ISSUE THREAD

**Issue by koralle** [yamada-ui/yamada-ui#1930]
### 1. Accessibility roles

There's nothing in particular to do.

### 2. `aria-controls`

* [x] Add an ID to `<div>` elements with the `.ui-accordion__panel` class.
* [x] Add the `aria-controls` attribute to <button> elements with the `.ui-accordion__label class`. Set the value of this attribute to the ID of the `<div>` elements added in the previous step.

![yamada-ui-accordion_1](https://github.com/yamada-ui/yamada-ui/assets/33865215/a8e748c4-920e-449f-8bc9-1c9b50d76f2f)

### 3. `aria-expanded`

* [x] Unnecessary `aria-expanded` attributes (on `li` element) should be removed.

![yamada-ui-accordion_1](https://github.com/yamada-ui/yamada-ui/assets/33865215/c8033bdc-9916-4b99-b869-437efc73254c)

### Keyboard Interaction

* [ ] `Enter`: activates the disclosure control and toggles the visibility of the disclosure content.
* [ ] `Space`: activates the disclosure control and toggles the visibility of the disclosure content.

### References

* [Disclosure (Show/Hide) Pattern | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/)
* [Example Disclosure (Show/Hide) for Image Description | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-image-description/)
* [Accordion - Yamada UI](https://yamada-ui.com/components/disclosure/accordion#use-accordionlabel)

### Problem Statement/Justification

We should prioritize web accessibility to ensure our UI components are usable by everyone, including people with disabilities. This inclusivity enhances user experience and complies with legal standards, broadening Yamada UI's user base.

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by cidkumagai** [comment]
I'll try it.
